Architecture_Engine/Create/Grid.cs

```diff
@@ -50,5 +50,22 @@ public static Grid Grid(Point origin, Vector direction, double length = 20)
         }
 
         /***************************************************/
+
+        public static Grid Grid(ICurve curve, string name)
+        {
+            return new Grid
+            {
+                Curve = Geometry.Modify.IProject(curve, Plane.XY),
+                Name = name,
+            };
+        }
+
+        /***************************************************/
+
+        public static Grid Grid(Point origin, Vector direction, string name, double length = 20)
```

Comment:
I would suggest having the name parameter being optional instead of having two create methods. 
I also think name should be the last parameter. I think this is the case for most create methods.

Reply:
As discussed - we cannot change the method signatures of existing methods because it would cause problems with existing scripts so we need to be careful. We could add `deprecated` tag but that can be done at any time later if that makes sense :smile: